c10/core/CPUProfilingAllocator.cpp

```diff
@@ -0,0 +1,411 @@
+#include <climits>
+
+#include <c10/core/CPUProfilingAllocator.h>
+
+namespace c10 {
+
+namespace {
+thread_local AllocationPlanner* allocation_planner{nullptr};
+thread_local CPUProfilingAllocator* profiling_allocator{nullptr};
+
+struct MemBlock {
+  uint64_t start_offset, end_offset;
+  MemBlock(uint64_t s, uint64_t e) : start_offset(s), end_offset(e) {}
+  bool operator<(const MemBlock& other) const {
+    return (start_offset < other.start_offset &&
+        end_offset <= other.start_offset);
```

Comment:
Shouldn't a block be considered non-overlapping if it is zero size, even if `start_offset == other.start_offset`? What sense is this `<` overload being used?

Reply:
If this is true, then the test here can be as simple as `end_offset <= other.start_offset`, right? (Assuming that `start_offset <= end_offset`)

Reply:
So this is used to make sure that allocation plan has not allocated overlapping regions. I am doing this by creating binary tree in which insertions are made. Two allocations are considered overlapping and thus will return false for `<` if when comparison is made for insertion.
I think your observation is correct. Thanks for commenting. I will fix that.